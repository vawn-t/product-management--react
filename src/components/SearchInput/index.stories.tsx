// Libraries
import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';

// Components
import SearchInput from './index';

// Styles
import '@assets/styles/App.css';
import '@assets/styles/variables.css';

// Constants

export default {
  title: 'Components/SearchInput',
  component: SearchInput
} as ComponentMeta<typeof SearchInput>;

const Template: ComponentStory<typeof SearchInput> = (args) => (
  <SearchInput {...args} />
);

export const Default = Template.bind({});
Default.args = {
  productName: 'Search value',
  onSearchProduct: () => undefined
};
