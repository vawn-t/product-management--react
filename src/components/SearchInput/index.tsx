// Library
import React, { memo } from 'react';

// Asset
import { searchIcon } from '@assets/index';

// Component
import { TextField } from '@components/index';

export interface SearchInputProps {
  productName: string;
  onSearchProduct: (value: string | number) => void;
}

const SearchInput: React.FC<SearchInputProps> = ({
  productName,
  onSearchProduct
}) => (
  <TextField
    placeholder='Searching...'
    iconUrl={searchIcon}
    height='4.25rem'
    iconWidth='2rem'
    onInputChange={onSearchProduct}
    defaultValue={productName}
  />
);

export default memo(SearchInput);
