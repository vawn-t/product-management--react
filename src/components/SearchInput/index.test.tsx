// Libraries
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { screen } from '@testing-library/dom';
import { create } from 'react-test-renderer';
import '@testing-library/jest-dom';

// Components
import SearchInput, { SearchInputProps } from './index';

describe('SearchInput component', () => {
  const defaultProps: SearchInputProps = {
    productName: 'Product Test',
    onSearchProduct: jest.fn()
  };

  let container: HTMLElement;
  beforeEach(() => {
    // setup a DOM element as a render target
    container = document.createElement('div');
    document.body.appendChild(container);
  });

  afterEach(() => {
    // cleanup on exiting
    container.remove();
    jest.clearAllMocks();
    cleanup();
  });

  test('should render correctly', () => {
    const defaultTree = create(<SearchInput {...defaultProps} />).toJSON();
    expect(defaultTree).toMatchSnapshot();
  });

  test('should correct value', () => {
    const productName = 'This is product testing';

    render(<SearchInput {...defaultProps} productName={productName} />);

    expect(screen.getByDisplayValue(productName)).toBeInTheDocument();
  });
});
