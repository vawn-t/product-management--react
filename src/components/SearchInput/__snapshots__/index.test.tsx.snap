// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput component should render correctly 1`] = `
<div
  className="field-wrapper"
>
  <div
    className="input-wrapper"
  >
    <input
      className="field field-standard"
      disabled={false}
      onChange={[Function]}
      placeholder="Searching..."
      readOnly={false}
      required={false}
      style={
        Object {
          "height": "4.25rem",
          "width": undefined,
        }
      }
      type="text"
      value="Product Test"
    />
  </div>
</div>
`;
